rust/lancedb/src/database/listing/manifest.rs

```diff
@@ -0,0 +1,3991 @@
+        &self,
+        object_id: String,
+        object_type: ObjectType,
+        location: Option<String>,
+        metadata: Option<String>,
+        base_objects: Option<Vec<String>>,
+    ) -> Result<()> {
+        let dataset_opt = self.manifest_dataset.lock().unwrap().clone();
+        let dataset = dataset_opt.ok_or_else(|| Error::Runtime {
+            message: "Manifest table not initialized".to_string(),
+        })?;
+
+        let schema = Self::manifest_schema();
+
+        // Create base_objects array from the provided list
+        use arrow_array::builder::{ListBuilder, StringBuilder};
+        let string_builder = StringBuilder::new();
+        let mut list_builder = ListBuilder::new(string_builder).with_field(Arc::new(Field::new(
+            "object_id",
+            DataType::Utf8,
+            true,
+        )));
+
+        match base_objects {
+            Some(objects) => {
+                for obj in objects {
+                    list_builder.values().append_value(obj);
+                }
+                list_builder.append(true);
+            }
+            None => {
+                list_builder.append_null();
+            }
+        }
+
+        let base_objects_array = list_builder.finish();
+
+        // Create arrays with optional values
+        let location_array = match location {
+            Some(loc) => Arc::new(StringArray::from(vec![Some(loc)])),
+            None => Arc::new(StringArray::from(vec![None::<String>])),
+        };
+
+        let metadata_array = match metadata {
+            Some(meta) => Arc::new(StringArray::from(vec![Some(meta)])),
+            None => Arc::new(StringArray::from(vec![None::<String>])),
+        };
+
+        let batch = RecordBatch::try_new(
+            schema.clone(),
+            vec![
+                Arc::new(StringArray::from(vec![object_id.as_str()])),
+                Arc::new(StringArray::from(vec![object_type.as_str()])),
+                location_array,
+                metadata_array,
+                Arc::new(base_objects_array),
+            ],
+        )
+        .map_err(|e| Error::Runtime {
+            message: format!("Failed to create manifest entry: {}", e),
+        })?;
+
+        let reader = Box::new(RecordBatchIterator::new(vec![Ok(batch)], schema.clone()));
+
+        // Use MergeInsert to ensure uniqueness on object_id
+        let dataset_guard = dataset.get().await?;
+        let dataset_arc = Arc::new(dataset_guard.clone());
+        drop(dataset_guard); // Drop read guard before merge insert
+
+        let mut merge_builder =
+            lance::dataset::MergeInsertBuilder::try_new(dataset_arc, vec!["object_id".to_string()])
+                .map_err(|e| Error::Lance { source: e })?;
+
+        // When matched: update all fields
+        merge_builder.when_matched(lance::dataset::WhenMatched::UpdateAll);
+        // When not matched: insert new row
+        merge_builder.when_not_matched(lance::dataset::WhenNotMatched::InsertAll);
```

Comment:
Hmm, merge_insert seems like the wrong way to have to solve this problem.  It's probably ok for now but it seems it would be a lot nicer if we could just add a "unique constraint" to the column.

Not something we need to fix right now.